src/test/java/io/confluent/connect/elasticsearch/integration/BaseConnectorIT.java

```diff
@@ -89,4 +91,29 @@ protected long waitForConnectorToStart(String name, int numTasks) throws Interru
       return Optional.empty();
     }
   }
+
+  public long waitForConnectorToFail(String name, int numTasks, String trace) throws InterruptedException {
+    TestUtils.waitForCondition(
+        () -> assertConnectorAndTasksFailed(name, numTasks, trace).orElse(false),
+        CONNECTOR_STARTUP_DURATION_MS,
+        "Connector tasks did not fail in time."
+    );
+    return System.currentTimeMillis();
+  }
+
+  protected Optional<Boolean> assertConnectorAndTasksFailed(String connectorName, int numTasks, String trace) {
```

Comment:
This logic is generic, is this available in some existing test libraries used in connectors?

Reply:
I don't think so.